### PR TITLE
Jamiecook/py311

### PR DESCRIPTION
**Pull request by jamiecook**

### CODE REVIEW

File: aequilibrae/paths/AoN.pyx [hunk 1]
```diff
@@ -13,7 +13,7 @@ include 'parallel_numpy.pyx'
 include 'path_file_saving.pyx'
 
 
-from .__version__ import binary_version as VERSION_COMPILED
+# from .__version__ import binary_version as VERSION_COMPILED
```
Comment (jamiecook) [inline aequilibrae/paths/AoN.pyx:16]: @pedrocamargo  - please to check

File: aequilibrae/paths/__init__.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 from aequilibrae.paths.traffic_assignment import TrafficAssignment
 from aequilibrae.paths.vdf import VDF
 from aequilibrae.paths.graph import Graph
-from .__version__ import binary_version, release_name, minor_version, release_version
+
+# from .__version__ import binary_version, release_name, minor_version, release_version
```
Comment (jamiecook) [inline aequilibrae/paths/__init__.py:12]:
```suggestion
```

File: aequilibrae/project/about.py [hunk 3]
```diff
@@ -35,10 +34,10 @@ def create(self):
             run_queries_from_sql_file(self.__conn, qry_file)
 
         cursor = self.__conn.cursor()
-        cursor.execute('select infovalue from about where infoname="aequilibrae_version"')
 
-        if cursor.fetchone()[0] is None:
-            cursor.execute(f"UPDATE 'about' set infovalue='{release_version}' where infoname='aequilibrae_version'")
+        sql = "SELECT count(*) as num_records from about;"
+        if self.__conn.execute(sql).fetchone()[0] == 0:
+            # cursor.execute(f"UPDATE 'about' set infovalue='{release_version}' where infoname='aequilibrae_version'")
```
Comment (jamiecook) [inline aequilibrae/project/about.py:40]: @pedrocamargo ... here too

```suggestion
```

File: aequilibrae/transit/gtfs/gtfs_sqlite_db.py [hunk 4]
```diff
@@ -74,6 +76,10 @@ def __init__(self, file_path, save_db, memory_db=False, spatialite_enabled=False
             self.logger.error(msg)
             raise ValueError(msg)
 
+        if self.save_db and self.memory_db:
```
Comment (jamiecook) [inline aequilibrae/transit/gtfs/gtfs_sqlite_db.py:79]: this one was reworked a little too.

File: tests/aequilibrae/project/test_osm_downloader.py [hunk 5]
```diff
@@ -14,33 +14,30 @@ def setUp(self) -> None:
...
-            thresh = 1.01
-
-        if random() < thresh:
-            self.o = OSMDownloader([[0.0, 0.0, 0.1, 0.1]], ["car"])
-            self.o.doWork()
-            if self.o.json:
-                self.fail("It found links in the middle of the ocean")
-        else:
-            print("Skipped check to not load OSM servers")
+        if not self.should_do_work():
```
Comment (jamiecook) [inline tests/aequilibrae/project/test_osm_downloader.py:17]: @pedrocamargo  double check this one too please

File: tests/aequilibrae/transit/test_create_gtfsdb.py [hunk 6]
```diff
@@ -24,9 +27,6 @@ def test_create_database(self):
         self.gtfs.conn.close()
 
     def test_import_gtfs(self):
-        # self.gtfs = create_gtfsdb(gtfs_folder, save_db=gtfs_db_output, overwrite=True, spatialite_enabled=True)
-        self.gtfs = create_gtfsdb(gtfs_folder, save_db=gtfs_db_output, overwrite=True, spatialite_enabled=False)
-        self.gtfs.import_gtfs()
-
-        self.gtfs = create_gtfsdb(gtfs_zip, save_db=gtfs_db_output, overwrite=True, spatialite_enabled=False)
+        self.gtfs = create_gtfsdb(gtfs_folder, save_db=None, spatialite_enabled=False, memory_db=True)
```
Comment (jamiecook) [inline tests/aequilibrae/transit/test_create_gtfsdb.py:30]: wasn't sure why it was happening twice, so made it only do once.

File: .github/workflows/build_mac.yml [hunk 7]
```diff
@@ -9,23 +9,36 @@ jobs:
...
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine
+        # export clang_dir=$(brew --prefix llvm@15)
+        # echo "Clang DIR: ${clang_dir}"
+        # export CC="${clang_dir}/bin/clang"
+        # export CXX="${clang_dir}/bin/clang++"
+        # export LDFLAGS="-L${clang_dir}/lib/c++ -Wl,-rpath,${clang_dir}/lib/c++"
+
```
Comment (jamiecook) [inline .github/workflows/build_mac.yml:29]:
```suggestion
```

File: aequilibrae/paths/AoN.pyx [hunk 8]
```diff
@@ -34,8 +34,8 @@ def one_to_all(origin, matrix, graph, result, aux_result, curr_thread):
 
     skims = len(graph.skim_fields)
 
-    if VERSION_COMPILED != graph.__version__:
-        raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
+    # if VERSION_COMPILED != graph.__version__:
+    #     raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
```
Comment (pedrocamargo) [inline aequilibrae/paths/AoN.pyx:38]:
```suggestion
```

File: aequilibrae/paths/AoN.pyx [hunk 9]
```diff
@@ -193,8 +193,8 @@ def path_computation(origin, destination, graph, results):
     if results.__graph_id__ != graph.__id__:
         raise ValueError("Results object not prepared. Use --> results.prepare(graph)")
 
-    if VERSION_COMPILED != graph.__version__:
-        raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
+    # if VERSION_COMPILED != graph.__version__:
+    #     raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
```
Comment (pedrocamargo) [inline aequilibrae/paths/AoN.pyx:197]:
```suggestion
```

File: aequilibrae/paths/AoN.pyx [hunk 10]
```diff
@@ -355,8 +355,8 @@ def skimming_single_origin(origin, graph, result, aux_result, curr_thread):
     if graph_fs[origin_index] == graph_fs[origin_index + 1]:
         raise ValueError("Centroid " + str(orig) + " does not exist in the graph")
 
-    if VERSION_COMPILED != graph.__version__:
-        raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
+    # if VERSION_COMPILED != graph.__version__:
+    #     raise ValueError('This graph was created for a different version of AequilibraE. Please re-create it')
```
Comment (pedrocamargo) [inline aequilibrae/paths/AoN.pyx:359]:
```suggestion
```

File: tests/aequilibrae/transit/test_create_gtfsdb.py [hunk 11]
```diff
@@ -9,12 +9,15 @@
...
     def setUp(self) -> None:
+        if os.path.isfile(gtfs_db_output):
+            os.unlink(gtfs_db_output)
         spatialite_folder = dirname(dirname(dirname(dirname(os.path.abspath(__file__)))))
         spatialite_folder = join(spatialite_folder, "aequilibrae/project")
         os.environ["PATH"] = f"{spatialite_folder};" + os.environ["PATH"]
 
     def tearDown(self) -> None:
         if os.path.isfile(gtfs_db_output):
+            print(gtfs_db_output)
```
Comment (pedrocamargo) [inline tests/aequilibrae/transit/test_create_gtfsdb.py:20]:
```suggestion
```

**Review by pedrocamargo — APPROVED** [review]

**Pull request merged**